### PR TITLE
Lenovo cnos user module

### PR DESCRIPTION
**Pull request by amuraleedhar**
##### SUMMARY
Introducing new module cnos_user to facilitate user configurations on Lenovo cnos devices. 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
lib/ansible/modules/network/cnos/cnos_user.py
test/integration/targets/cnos_user/aliases
test/integration/targets/cnos_user/cnos_user_sample_hosts
test/integration/targets/cnos_user/defaults/main.yaml
test/integration/targets/cnos_user/tasks/cli.yaml
test/integration/targets/cnos_user/tasks/main.yaml
test/integration/targets/cnos_user/tests/common/basic.yaml
test/integration/targets/cnos_user/tests/common/sanity.yaml
test/units/modules/network/cnos/fixtures/cnos_user_config.cfg
test/units/modules/network/cnos/test_cnos_user.py

##### ANSIBLE VERSION
ansible 2.8.0.dev0 (devel e2b9c36080) last updated 2018/10/12 13:58:41 (GMT +550)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/ansible/sheru/ansible/lib/ansible
  executable location = /home/ansible/sheru/ansible/bin/ansible
  python version = 2.7.6 (default, Nov 23 2017, 15:49:48) [GCC 4.8.4]

##### ADDITIONAL INFORMATION
This has been tested with G8272 CNOS Switch of Lenovo version 10.9.1 version.

### CODE REVIEW

**Comment by amuraleedhar** [comment]
This is complete from my side. Please review.

File: lib/ansible/modules/network/cnos/cnos_user.py [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+
+from ansible.module_utils.network.cnos.cnos import run_commands, load_config
+from ansible.module_utils.network.cnos.cnos import get_config
+from ansible.module_utils.network.cnos.cnos import cnos_argument_spec
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.six import string_types, iteritems
+from ansible.module_utils.network.common.utils import to_list
+from ansible.module_utils.network.common.utils import remove_default_spec
+
+VALID_ROLES = ['network-admin', 'network-operator']
```
Comment (justjais) [inline lib/ansible/modules/network/cnos/cnos_user.py]: This constant should ideally be defined in `module.utils` code and imported. So, that in future if any resource module uses the same constant or want to extend the same can use it without duplicating.

File: test/integration/targets/cnos_user/tests/common/sanity.yaml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+  - name: remove idempotency
+    cnos_user: *remove
+    register: result
+
+  - assert: *false
+
+  - debug: msg="skipping sshkey test as the key needs to be created on the server first"
+
+#  - name: create a new user
```
Comment (justjais) [inline test/integration/targets/cnos_user/tests/common/sanity.yaml]: Do you need to have these commented test cases in the test.

File: test/integration/targets/cnos_user/tests/common/sanity.yaml [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    - assert: *true
+
+#    when: idem
+
+  - name: tearDown
+    cnos_user: &tear
+      name: ansible
+      purge: yes
+#      provider: "{{ connection }}"
```
Comment (justjais) [inline test/integration/targets/cnos_user/tests/common/sanity.yaml]: please remove the commented line if not added for any specific purpose.

File: test/integration/targets/cnos_user/tests/common/sanity.yaml (hunk 2 above)
Comment (amuraleedhar) [reply test/integration/targets/cnos_user/tests/common/sanity.yaml]: I have removed them. Thanks

File: test/integration/targets/cnos_user/tests/common/sanity.yaml (hunk 3 above)
Comment (amuraleedhar) [reply test/integration/targets/cnos_user/tests/common/sanity.yaml]: Removed. Thanks 
-Anil

File: lib/ansible/modules/network/cnos/cnos_user.py (hunk 1 above)
Comment (amuraleedhar) [reply lib/ansible/modules/network/cnos/cnos_user.py]: Moved to module_utils/ cnos.py

**Review by justjais — APPROVED** [review]

**Pull request merged**